config/default/forms/app/pregnancy.xml

```diff
@@ -978,105 +978,105 @@
+      <bind constraint="(. &gt;= today()) and
+(decimal-date-time(.) &lt;= decimal-date-time(today()) + 30)" jr:constraintMsg="jr:itext('/pregnancy/anc_visits_hf/anc_visits_hf_next/anc_next_visit_date/appointment_date:jr:constraintMsg')" nodeset="/pregnancy/anc_visits_hf/anc_visits_hf_next/anc_next_visit_date/appointment_date" relevant="selected(../appointment_date_known, 'yes')" required="true()" type="date"/>
       <bind nodeset="/pregnancy/anc_visits_hf/anc_visits_hf_next/anc_visit_advice_note" relevant="selected(../anc_next_visit_date/appointment_date_known, 'no')"/>
       <bind nodeset="/pregnancy/anc_visits_hf/anc_visits_hf_next/anc_visit_advice_note/who_recommends_note" readonly="true()" type="string"/>
       <bind nodeset="/pregnancy/anc_visits_hf/anc_visits_hf_next/anc_visit_advice_note/prenancy_age_note" readonly="true()" type="string"/>
       <bind nodeset="/pregnancy/anc_visits_hf/anc_visits_hf_next/anc_visit_advice_note/refer_note" readonly="true()" type="string"/>
-      <bind nodeset="/pregnancy/risk_factors/risk_factors_history/first_pregnancy" type="select1" required="true()"/>
-      <bind nodeset="/pregnancy/risk_factors/risk_factors_history/previous_miscarriage" type="select1" required="true()"/>
-      <bind nodeset="/pregnancy/risk_factors/risk_factors_present/primary_condition" type="select" required="true()" relevant="selected(../../risk_factors_history/first_pregnancy, 'yes')" jr:constraintMsg="jr:itext('/pregnancy/risk_factors/risk_factors_present/primary_condition:jr:constraintMsg')" constraint="not(selected(.,'none')) or count-selected(.) &lt; 2"/>
-      <bind nodeset="/pregnancy/risk_factors/risk_factors_present/secondary_condition" type="select" required="true()" relevant="selected(../../risk_factors_history/first_pregnancy, 'no')" jr:constraintMsg="jr:itext('/pregnancy/risk_factors/risk_factors_present/secondary_condition:jr:constraintMsg')" constraint="not(selected(.,'none')) or count-selected(.) &lt; 2"/>
-      <bind nodeset="/pregnancy/risk_factors/risk_factors_present/additional_risk_check" type="select1" required="true()"/>
-      <bind nodeset="/pregnancy/risk_factors/risk_factors_present/additional_risk" type="string" required="true()" relevant="selected(../additional_risk_check, 'yes')" jr:constraintMsg="jr:itext('/pregnancy/risk_factors/risk_factors_present/additional_risk:jr:constraintMsg')" constraint="string-length(.) &lt;= 100"/>
-      <bind nodeset="/pregnancy/risk_factors/r_risk_factor_present" type="string" calculate="if(selected(../risk_factors_history/first_pregnancy, 'yes') or
+      <bind nodeset="/pregnancy/risk_factors/risk_factors_history/first_pregnancy" required="true()" type="select1"/>
+      <bind nodeset="/pregnancy/risk_factors/risk_factors_history/previous_miscarriage" relevant="selected(../first_pregnancy, 'no')" required="true()" type="select1"/>
```

Comment:
@Benmuiruri can you confirm if this was the only actual change to the form logic?  I think so, based on context and a brief skim of the rest of the diff, but just want to be sure...

(Also JSYK, this whole mess of changes within the form xml file because of indeterminate arrangement of the attributes will be fixed by updating our pyxform version (whenever we get around to that...).  Until then, I typically try to only commit the xml lines that have actually been updated (though sometimes this is not feasible). No worries on this PR, but something to consider in the future.)

Reply:
Thanks for the Info Josh.